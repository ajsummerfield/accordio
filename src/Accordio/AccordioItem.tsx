import React from 'react';
import AccordioItemContext from './context/AccordioItemContext';
import useAccordioContext from './hooks/useAccordioContext';
import useAccordioItemContext from './hooks/useAccordioItemContext';

type AccordioItemProps = {
  index: number;
  children: JSX.Element;
};

const AccordioItem = ({ index, children }: AccordioItemProps) => {
  const { activeIndex, setActiveIndex } = useAccordioContext();
  const { setIndex } = useAccordioItemContext();
  const className = `accordio__item 
    ${activeIndex === index ? 'accordio__item--active' : ''}`;

  setIndex(index);

  const handleOnClick = () => {
    setActiveIndex(activeIndex !== index ? index : null);
  };

  const context = {
    index: index,
    setIndex: setIndex,
  };

  return (
    <AccordioItemContext.Provider value={context}>
      <div className={className} onClick={handleOnClick}>
        {children}
      </div>
    </AccordioItemContext.Provider>
  );
};

export default AccordioItem;
