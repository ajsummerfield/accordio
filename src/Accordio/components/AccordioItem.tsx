import React, { useEffect } from 'react';
import AccordioItemContext from '../context/AccordioItemContext';
import useAccordioContext from '../hooks/useAccordioContext';
import useAccordioItemContext from '../hooks/useAccordioItemContext';

type AccordioItemProps = {
  index: number;
  children: any;
};

const AccordioItem = ({ index, children }: AccordioItemProps) => {
  const { activeIndex, setActiveIndex } = useAccordioContext();
  const { setIndex } = useAccordioItemContext();
  const className = `accordio__item 
    ${activeIndex === index ? 'accordio__item--active' : ''}`;

  useEffect(() => {
    setIndex(index);
  }, [index]);

  const handleOnClick = () => {
    setActiveIndex(activeIndex !== index ? index : -1);
  };

  const context = {
    index: index,
    setIndex: setIndex,
  };

  return (
    <AccordioItemContext.Provider value={context}>
      <div className={className} onClick={handleOnClick}>
        {children}
      </div>
    </AccordioItemContext.Provider>
  );
};

export default AccordioItem;
