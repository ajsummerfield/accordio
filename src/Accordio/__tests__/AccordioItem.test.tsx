import React from 'react';
import AccordioItem from '../AccordioItem';
import { shallowRenderer } from './helpers';

describe('AccordioItem Component', () => {
  let component: JSX.Element;

  beforeEach(() => {
    component = <AccordioItem index={0}>Hello World</AccordioItem>;
  });

  it('Should match the snapshot', () => {
    const tree = shallowRenderer(component);
    expect(tree).toMatchSnapshot();
  });
});
