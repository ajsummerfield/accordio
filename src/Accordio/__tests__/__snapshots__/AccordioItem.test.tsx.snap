// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordioItem Component Should match the snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "index": 0,
      "setIndex": [Function],
    }
  }
>
  <div
    className="accordio__item 
    "
    onClick={[Function]}
  >
    Hello World
  </div>
</Context.Provider>
`;
